### PR TITLE
Changes for top level taxId: Bsd 1082

### PR DESCRIPTION
**Pull request by danielvaughan**
These are the changes for a top level taxId. Do we want to merge this so we can run the ena-pipeline in wwwdev and give ENA some new examples with this included?

### CODE REVIEW

File: models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java [hunk 1]
```diff
@@ -1,503 +1,516 @@
...
+        return ZonedDateTime.ofInstant(release, ZoneOffset.UTC).format(ISO_LOCAL_DATE);
+    }
+
+    @JsonProperty(value = "updateDate", access = JsonProperty.Access.READ_ONLY)
+    public String getUpdateDate() {
+        return ZonedDateTime.ofInstant(update, ZoneOffset.UTC).format(ISO_LOCAL_DATE);
+    }
+
+    @JsonProperty(value = "taxId", access = JsonProperty.Access.READ_ONLY)
+    public Integer[] getTaxId() {
```
Comment (Kerruba) [inline models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: Out of curiosity there's a reason to prefer an Integer[] compared to a List<Integer>?

File: models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java [hunk 2]
```diff
@@ -1,503 +1,516 @@
...
+        List<Integer> taxIds = new ArrayList<>();
+        for (Attribute attribute : attributes) {
```
Comment (Kerruba) [inline models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: Why not using streams all the way through?
``` java
return attributes.parallelStream()
.filter(attr -> attr.getType().toLowerCase().equalsIgnoreCase("Organism") && !attr.getIri().isEmpty())
.map(this::getIri) // or even flatMap if more than one IRI is returned
.map(Object::toString)
.map(this::extractTaxIdFromIri)
.collect(Collectors.toList())
```

Comment (danielvaughan) [reply models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: Great point, thanks

File: models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java (hunk 1 above)
Comment (danielvaughan) [reply models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: I don't know, yes, that is strange

File: models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java (hunk 2 above)
Comment (danielvaughan) [reply models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: Hmm this fails to compile:
```
        return attributes.stream()
                .filter(attr -> attr.getType().toLowerCase().equalsIgnoreCase("Organism") && !attr.getIri().isEmpty())
                .map(attr -> attr::getIri) // or even flatMap if more than one IRI is returned
                .map(Object::toString)
                .map(this::extractTaxIdFromIri)
                .collect(Collectors.toList());
```
attr::getIri "Object is not a functional interface"

Comment (Kerruba) [reply models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: Try with
```java
Attribute::getIri
```

File: models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java (hunk 1 above)
Comment (danielvaughan) [reply models/core/src/main/java/uk/ac/ebi/biosamples/model/Sample.java]: There is an error if I use List<Integer>. The serialization test fails apparently as it is trying to set taxId

**Comment by Kerruba** [comment]
@danielvaughan have you checked what happens if someone submit a samples with a taxID? Is returning an error or is submitting the sample without reading that information?

**Comment by Kerruba** [comment]
Can we merge this into master? Anything else you would like to check?

**Pull request merged**